### ISSUE TITLE
AttributeError: 'str' object has no attribute 'url'

### ISSUE THREAD

**Issue by cgimenes** [archlinux/archinstall#2991]
### Which ISO version are you using?

2024-12-01

### The installation log

```json
https://0x0.st/XRh8.log
```

#### Description of the issue

I've got this error while trying to use an older configuration file (version 2.6.3): https://github.com/cgimenes/dotfiles/blob/938b9ca465950655df89aa7786bf97fba04c1bff/setup/arch/user_configuration.json

But, I also tried to install without using any configuration file and the error persisted.

**Comment by iTzVirtual00** [comment]
Same error here, in my case _it seems_ to be a problem with mirror setting serialization, it serializes the {"url":...} as string instead of json:
```
"Italy": [
	"{\"url\":\"http://it.mirrors.cicku.me/archlinux/\",\"protocol\":\"http\",\"active\":true,\"country\":\"Italy\",\"country_code\":\"IT\",\"is>
	"{\"url\":\"https://it.mirrors.cicku.me/archlinux/\",\"protocol\":\"https\",\"active\":true,\"country\":\"Italy\",\"country_code\":\"IT\",\">
	"{\"url\":\"http://archlinux.mirror.garr.it/archlinux/\",\"protocol\":\"http\",\"active\":true,\"country\":\"Italy\",\"country_code\":\"IT\">
	"{\"url\":\"http://archlinux.mirror.server24.net/\",\"protocol\":\"http\",\"active\":true,\"country\":\"Italy\",\"country_code\":\"IT\",\"is>
	"{\"url\":\"https://archlinux.mirror.server24.net/\",\"protocol\":\"https\",\"active\":true,\"country\":\"Italy\",\"country_code\":\"IT\",\">
	"{\"url\":\"https://arch.mirror.hyperbit.it/\",\"protocol\":\"https\",\"active\":true,\"country\":\"Italy\",\"country_code\":\"IT\",\"isos\">
]
```
I deleted the "mirror_config" body from the json as a workaround.

**Comment by svartkanin** [comment]
I had a look and this is a bigger bug than expected. I'll try to have a look at it over the weekend

**Comment by msrsaditya** [comment]
> https://github.com/cgimenes/dotfiles/blob/938b9ca465950655df89aa7786bf97fba04c1bff/setup/arch/user_configuration.json

I also have the same error -- I'm using this config file "https://raw.githubusercontent.com/msrsaditya/Arch-Cinnamon/refs/heads/main/Installation/user_configuration.json"

Should I just remove the "mirror_config" body for now? Is there anything else I can do?

**Comment by svartkanin** [comment]
Yes, you can remove it from the config and then manually set it in the menu system.

**Comment by msrsaditya** [comment]
> Yes, you can remove it from the config and then manually set it in the menu system.

Thanks for all the hard work. I'm hoping this would get resolved quickly atleast by the next ISO release in JAN.

**Comment by svartkanin** [comment]
Not to smash hopes but potentially not. It's holiday period so things move slower

**Comment by svartkanin** [comment]
A fix is up, if someone would like to try it either clone the PR https://github.com/archlinux/archinstall/pull/3007 or use the build image from the PR

**Comment by svartkanin** [comment]
It's merged into master

**Comment by nikelborm** [comment]
Sadly, today I faced the same issue 2 months later since a fix was made. How often is archinstall being updated in Arch ISOs?

**Comment by svartkanin** [comment]
This is included in the latest release 3.0.2

**Comment by nikelborm** [comment]
Yes, I understand that the fix is in the version `3.0.2` which was released after `3.0.1` (the one that was included in ISO).

The thing I didn't understand initially is that `3.0.2` was released only <ins>4 days ago</ins> when I was assuming it was released in December meaning before the `01.01.2025` Arch ISO release. Based on that false assumption I was trying to figure out why `3.0.2` didn't get into `01.01.2025` Arch ISO.

Now I understand why.
